core/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -567,11 +584,13 @@ private InternalAggregations reduceAggs(List<InternalAggregations> aggregationsL
         final int from;
         // sort value formats used to sort / format the result
         final DocValueFormat[] sortValueFormats;
+        // The fields that were flagged as unmapped by all indices touched in this request
+        final Set<String> unanimouslyUnmappedFieldNames;
 
         ReducedQueryPhase(long totalHits, long fetchHits, float maxScore, boolean timedOut, Boolean terminatedEarly, Suggest suggest,
                           InternalAggregations aggregations, SearchProfileShardResults shardResults, ScoreDoc[] scoreDocs,
                           SortField[] sortFields, DocValueFormat[] sortValueFormats, int numReducePhases, boolean isSortedByField, int size,
-                          int from, boolean isEmptyResult) {
+                          int from, boolean isEmptyResult, Set<String> unanimouslyUnmappedFieldNames) {
```

Comment:
I see many places that need to null check this variable. Can we require that it is not null all the time and make old indices just pass `Collections.emptySet()`?